app/src/main/java/org/p2p/wallet/solend/interactor/SolendDepositsInteractor.kt

```diff
@@ -0,0 +1,24 @@
+package org.p2p.wallet.solend.interactor
+
+import org.p2p.wallet.solend.repository.SolendDepositsRepository
+import org.p2p.wallet.solend.model.SolendDepositToken
+import org.p2p.wallet.solend.model.SolendDomainMapper
+import org.p2p.wallet.user.repository.UserLocalRepository
+
+class SolendDepositsInteractor(
+    private val repository: SolendDepositsRepository,
+    private val userLocalRepository: UserLocalRepository,
+    private val mapper: SolendDomainMapper
+) {
+
+    suspend fun getAllDeposits(tokenSymbols: List<String>): List<SolendDepositToken> {
+        val marketsInfo = repository.getSolendMarketInfo(tokenSymbols)
+        val activeDeposits = repository.getDeposits()
+
+        return marketsInfo.mapNotNull { info ->
+            val tokenData = userLocalRepository.findTokenDataBySymbol(info.tokenSymbol) ?: return@mapNotNull null
+            val activeDeposit = activeDeposits.find { it.depositTokenSymbol == info.tokenSymbol }
+            mapper.toDepositToken(tokenData, info, activeDeposit)
```

Comment:
interactor is not responsible for mapping network entities to domain entities
put mapping in the SolendDepositsRepository

Reply:
This is not network entities, these are merging several domain entities to single domain entity. 
And I was really struggling about the place of mapping. 

I am not sure if we need to make the repository method be complex with several remote requests. 

What do you think guys?

Reply:
i think if domain requests complex domain entity - it's repository responsibility to make all the needed requests and build the entity.
just like with tables - if we want multiple complex data - the repository makes all the requests in database and build it in the mapper

Reply:
okay, let me try to move it in repo

Reply:
done